File: src/main/java/cardgame/CardDeck.java
package cardgame;

public enum CardDeck {
	INSTANCE;

	private final char[] deck;

	private CardDeck() {
		this.deck = new char[] { '2', '3', '4', '5', '6', '7', '8', '9', 'T', 'J' };
	}

	public static char get(int number) {
		return INSTANCE.deck[number % INSTANCE.deck.length];
	}

	public static int valueOf(char c) {
		return new String(INSTANCE.deck).indexOf(c);
	}
}

File: carddeck.js

module.exports = class CardDeck {

	constructor() {
		this.deck = "23456789TJQKA";
	}

	static get(number) {
        let cardDeck = new CardDeck();
        return cardDeck.deck.charAt(number % cardDeck.deck.length);
	}

	static valueOf(char) {
		return new CardDeck().deck.indexOf(char);
	}
}
